From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Wirāmu Pauling <aenertia@aenertia.net>
Date: Thu, 05 Mar 2026 14:00:00 +1300
Subject: [PATCH] arm64: dts: rockchip: rk356x: add CPU energy model to
 remaining cores and thermal sustainable-power

Mainline 6.18 added dynamic-power-coefficient=138 to cpu0 but not
cpu1-3. Add it to all remaining cores for complete Energy Aware
Scheduling support. Add sustainable-power to the cpu_thermal zone
for informed thermal throttling decisions.

Values validated against FNB58 inline power profiling on RGB30.

Signed-off-by: Joel Wirāmu Pauling <aenertia@aenertia.net>
---
 arch/arm64/boot/dts/rockchip/rk356x-base.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi b/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi
index fd2214b6fad4..a1b2c3d4e5f6 100644
--- a/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x-base.dtsi
@@ -71,6 +71,7 @@
 			compatible = "arm,cortex-a55";
 			reg = <0x0 0x100>;
 			#cooling-cells = <2>;
+			dynamic-power-coefficient = <138>;
 			cpu-idle-states = <&CPU_SLEEP>;
 			enable-method = "psci";
 			i-cache-size = <0x8000>;
@@ -87,6 +88,7 @@
 			compatible = "arm,cortex-a55";
 			reg = <0x0 0x200>;
 			#cooling-cells = <2>;
+			dynamic-power-coefficient = <138>;
 			cpu-idle-states = <&CPU_SLEEP>;
 			enable-method = "psci";
 			i-cache-size = <0x8000>;
@@ -103,6 +105,7 @@
 			compatible = "arm,cortex-a55";
 			reg = <0x0 0x300>;
 			#cooling-cells = <2>;
+			dynamic-power-coefficient = <138>;
 			cpu-idle-states = <&CPU_SLEEP>;
 			enable-method = "psci";
 			i-cache-size = <0x8000>;
@@ -1569,6 +1572,7 @@
 		cpu_thermal: cpu-thermal {
 			polling-delay-passive = <100>;
 			polling-delay = <1000>;
+			sustainable-power = <1000>;

 			thermal-sensors = <&tsadc 0>;

